```log
=== RUN   TestAccAliCloudRdsDBInstance_Mysql_8_0--- PASS: TestAccAliCloudRdsDBInstance_Mysql_8_0 (2906.00s)
=== RUN   TestAccAliCloudRdsDBInstancePostgreSQL--- PASS: TestAccAliCloudRdsDBInstancePostgreSQL (3588.84s)
=== RUN   TestAccAliCloudRdsDBInstance_PG_MajorVersionUpgrade_Prepaid--- PASS: TestAccAliCloudRdsDBInstance_PG_MajorVersionUpgrade_Prepaid (2947.29s)
=== RUN   TestAccAliCloudRdsDBInstance_VpcId--- PASS: TestAccAliCloudRdsDBInstance_VpcId (1608.28s)
=== RUN   TestAccAliCloudRdsDBInstance_Mysql_5_7_HighAvailabilityInstance--- PASS: TestAccAliCloudRdsDBInstance_Mysql_5_7_HighAvailabilityInstance (1341.25s)
=== RUN   TestAccAliCloudRdsDBInstance_SQLServer--- PASS: TestAccAliCloudRdsDBInstance_SQLServer (2683.14s)
=== RUN   TestAccAliCloudRdsDBInstance_PostgreSQL_12_0--- PASS: TestAccAliCloudRdsDBInstance_PostgreSQL_12_0 (2984.83s)
=== RUN   TestAccAliCloudRdsDBInstance_PostgreSQL_13_0_SSL--- PASS: TestAccAliCloudRdsDBInstance_PostgreSQL_13_0_SSL (3646.03s)
=== RUN   TestAccAliCloudRdsDBInstance_MariaDB_10_3--- PASS: TestAccAliCloudRdsDBInstance_MariaDB_10_3 (3265.54s)
=== RUN   TestAccAliCloudRdsDBInstance_Mysql_8_0_MultiAZ--- PASS: TestAccAliCloudRdsDBInstance_Mysql_8_0_MultiAZ (720.34s)
=== RUN   TestAccAliCloudRdsDBInstanceMysql_DBEncryptionKey--- PASS: TestAccAliCloudRdsDBInstanceMysql_DBEncryptionKey (1027.98s)
=== RUN   TestAccAliCloudRdsDBInstanceMysql_DBParamGroupId--- PASS: TestAccAliCloudRdsDBInstanceMysql_DBParamGroupId (938.82s)
=== RUN   TestAccAliCloudRdsDBInstanceMysql_general_essd--- PASS: TestAccAliCloudRdsDBInstanceMysql_general_essd (1858.08s)
=== RUN   TestAccAliCloudRdsDBInstance_PostgreSQL_15_0_Babelfish--- FAIL: TestAccAliCloudRdsDBInstance_PostgreSQL_15_0_Babelfish (7.88s)
=== RUN   TestAccAliCloudRdsDBInstance_Mysql_8_0_PrePaid--- PASS: TestAccAliCloudRdsDBInstance_Mysql_8_0_PrePaid (892.20s)
=== RUN   TestAccAliCloudRdsDBInstance_Mysql_8_0_Cluster--- PASS: TestAccAliCloudRdsDBInstance_Mysql_8_0_Cluster (968.63s)
=== RUN   TestAccAliCloudRdsDBInstance_MySQL_8_0_ServerlessBasic--- PASS: TestAccAliCloudRdsDBInstance_MySQL_8_0_ServerlessBasic (1108.73s)
=== RUN   TestAccAliCloudRdsDBInstance_PostgreSQL_14_0_Serverless--- PASS: TestAccAliCloudRdsDBInstance_PostgreSQL_14_0_Serverless (1498.94s)
=== RUN   TestAccAliCloudRdsDBInstanceMySQL_ServerlessStandard--- PASS: TestAccAliCloudRdsDBInstanceMySQL_ServerlessStandard (696.14s)
=== RUN   TestAccAliCloudRdsDBInstance_SQLServer_2019_ServerlessHA--- SKIP: TestAccAliCloudRdsDBInstance_SQLServer_2019_ServerlessHA (0.00s)
=== RUN   TestAccAliCloudRdsDBInstanceMysql_Downgrade--- PASS: TestAccAliCloudRdsDBInstanceMysql_Downgrade (1140.02s)
=== RUN   TestAccAliCloudRdsDBInstanceMysql_general_essd_supplement--- PASS: TestAccAliCloudRdsDBInstanceMysql_general_essd_supplement (748.08s)
=== RUN   TestAccAliCloudRdsDBInstanceSqlService_general_essd--- PASS: TestAccAliCloudRdsDBInstanceSqlService_general_essd (727.56s)
```

--- a/alicloud/resource_alicloud_db_instance.go
+++ b/alicloud/resource_alicloud_db_instance.go
@@ -811,7 +811,7 @@ func resourceAliCloudDBInstanceUpdate(d *schema.ResourceData, meta interface{}) 
 		err = resource.Retry(d.Timeout(schema.TimeoutUpdate), func() *resource.RetryError {
 			response, err = client.RpcPost("Rds", "2014-08-15", action, nil, request, false)
 			if err != nil {
-				if NeedRetry(err) {
+				if NeedRetry(err) || IsExpectedErrors(err, []string{"OperationDenied.DBInstanceStatus"}) {
 					wait()
 					return resource.RetryableError(err)
 				}
@@ -1094,11 +1094,11 @@ func resourceAliCloudDBInstanceUpdate(d *schema.ResourceData, meta interface{}) 
 			return WrapErrorf(err, DefaultErrorMsg, d.Id(), action, AlibabaCloudSdkGoERROR)
 		}
 		addDebug(action, response, request)
-		stateConf = BuildStateConf([]string{}, []string{"Running"}, d.Timeout(schema.TimeoutCreate), 3*time.Minute, rdsService.RdsDBInstanceStateRefreshFunc(d.Id(), []string{"Deleting"}))
+		stateConf := BuildStateConf([]string{}, []string{"Success"}, d.Timeout(schema.TimeoutUpdate), 3*time.Second, rdsService.RdsUpgradeMajorVersionRefreshFunc(d.Id(), formatInt(response["TaskId"]), []string{"Failed"}))
 		if _, err := stateConf.WaitForState(); err != nil {
 			return WrapErrorf(err, IdMsg, d.Id())
 		}
-		stateConf := BuildStateConf([]string{}, []string{"Success"}, d.Timeout(schema.TimeoutUpdate), 3*time.Second, rdsService.RdsUpgradeMajorVersionRefreshFunc(d.Id(), formatInt(response["TaskId"]), []string{"Failed"}))
+		stateConf = BuildStateConf([]string{}, []string{"Running"}, d.Timeout(schema.TimeoutCreate), 3*time.Minute, rdsService.RdsDBInstanceStateRefreshFunc(d.Id(), []string{"Deleting"}))
 		if _, err := stateConf.WaitForState(); err != nil {
 			return WrapErrorf(err, IdMsg, d.Id())
 		}
@@ -1133,7 +1133,7 @@ func resourceAliCloudDBInstanceUpdate(d *schema.ResourceData, meta interface{}) 
 		err = resource.Retry(d.Timeout(schema.TimeoutUpdate), func() *resource.RetryError {
 			response, err = client.RpcPost("Rds", "2014-08-15", action, nil, request, true)
 			if err != nil {
-				if NeedRetry(err) {
+				if NeedRetry(err) || IsExpectedErrors(err, []string{"IncorrectDBInstanceState"}) {
 					wait()
 					return resource.RetryableError(err)
 				}

--- a/website/docs/r/db_instance.html.markdown
+++ b/website/docs/r/db_instance.html.markdown
@@ -707,7 +707,7 @@ The following arguments are supported:
   - Before: The system collects the statistics of the instance before the switchover to ensure service stability. If the instance contains a large amount of data, the upgrade may require a long period of time.
   - After: The system collects the statistics of the instance after the switchover to accelerate the upgrade. After the upgrade, if you access tables for which no statistics are generated, the query plans may be inaccurate, and your database service may be unavailable during peak hours.
 
-    -> **NOTE:** If you set the SwitchOver parameter to false, the value Before specifies that the system collects the statistics of the instance before the instance starts to process read and write requests, and the value After specifies that the system collects the statistics of the instance after the instance starts to process read and write requests.
+    -> **NOTE:** This parameter is required when performing a major version upgrade (`engine_version` changes), because RDS for PostgreSQL only supports local upgrade. The value Before specifies that the system collects the statistics of the instance before the instance starts to process read and write requests, and the value After specifies that the system collects the statistics of the instance after the instance starts to process read and write requests.
 
 * `collation` - (Optional, Available since v1.277.0) The character set collation of the instance. By default, the system does not modify the character set collation of the instance. Valid values:
   - `Chinese_PRC_CI_AS`

